src/Core/Settings.cpp

```diff
@@ -7037,6 +7037,9 @@ Possible values:
 - positive integer - max length of the name,
 - 0 — always,
 - negative integer - never.
+)", 0) \
+    DECLARE(UInt64, max_subcolumns, 10000, R"(
```

Comment:
New settings should be added to file `SettingsChangesHistory.cpp` where we keep track of changes in settings.